tm2/pkg/bft/state/store.go

```diff
@@ -131,8 +132,13 @@ type ABCIResponses struct {
 
 // NewABCIResponses returns a new ABCIResponses
 func NewABCIResponses(block *types.Block) *ABCIResponses {
-	resDeliverTxs := make([]abci.ResponseDeliverTx, block.NumTxs)
-	if block.NumTxs == 0 {
+	return NewABCIResponsesFromNum(block.NumTxs)
+}
+
+// NewABCIResponsesFromNum returns a new ABCIResponses with a set number of txs
+func NewABCIResponsesFromNum[N constraints.Integer](numTxs N) *ABCIResponses {
```

Comment:
couldn't we just use `int`, here?

Reply:
there is two places where this is used, one is int64, one is int
we could use a cast a call site but I thought this would be better, tell me if you want me to change this

Reply:
please, use a cast call and keep things simple.